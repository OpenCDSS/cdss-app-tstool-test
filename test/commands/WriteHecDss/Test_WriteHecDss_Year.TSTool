# Write Yearly data to DSS
#@os Windows
# Test setup...
StartLog(LogFile="Results\Test_WriteHecDss_Year.TSTool.log")
# SetDebugLevel(LogFileLevel=10)
# Write a single blank time series to the DSS file to allow WriteHecDss to be closed; allows deleting DSS file.
NewTimeSeries(Alias="Blank",NewTSID="Blank...Month",SetStart="1960-01",SetEnd="1960-12",Units="CFS")
WriteHecDss(OutputFile="Results\Test_WriteHecDss_Year_out.dss",Type=PER-AVER,Close=True)
Free(TSList=AllMatchingTSID,TSID="*")
RemoveFile(InputFile="Results\Test_WriteHecDss_Year_out.dss")
RemoveFile(InputFile="Results\Test_WriteHecDss_Year.dssutl.out",IfNotFound=Ignore)
# The generated time series has identifier parts that indicate the HEC-DSS
# parts, other than the D part, which is related to the period of record.
NewPatternTimeSeries(Alias="Alias1",NewTSID="Apart1:Bpart1-with-dashes..Cpart1.Year.Fpart1",Description="Test data",SetStart="1950",SetEnd="1975",Units="MGD",PatternValues="5,10,12,13,75")
WriteHecDss(OutputFile="Results/Test_WriteHecDss_Year_out.dss",Type=PER-CUM)
#
# Use DSSUTL to output the data from the resulting DSS file to compare results
RunDSSUTL(DssFile="Results\Test_WriteHecDss_Year_out.dss",InputFile="Test_WriteHecDss_Year.MCO",DssutlProgram="${WorkingDir}\DSSUTL.EXE")
# In order to regenerate the expected results file, edit the output file in the file: Test_WriteHecDss_Year.MCO
CompareFiles(InputFile1="ExpectedResults\Test_WriteHecDss_Year.dssutl.out",InputFile2="Results\Test_WriteHecDss_Year.dssutl.out",IfDifferent=Warn)
