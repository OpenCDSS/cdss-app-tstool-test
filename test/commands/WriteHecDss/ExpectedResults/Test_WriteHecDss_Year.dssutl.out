  
 /APART1/BPART1-WITH-DASHES/CPART1/01JAN1900/1YEAR/FPART1/
 Start: 31DEC1950 at 2400 hours;   End: 31DEC1975 at 2400 hours;  Number:   26
 Units: MGD        Type: PER-CUM 
 31DEC1950, 2400;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 31DEC1957, 2400;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 31DEC1964, 2400;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 31DEC1971, 2400;    10.00   12.00   13.00   75.00    5.00
 ----------------------------------------------------------------------------
  
